FrameworkInternals/DesignInspector.py

```diff
@@ -110,24 +110,25 @@ def class_has_legit_device_parent(self, class_name):
             return False
         return True
 
-    def get_parent_struct(self, class_name):
-        """ Returns whatever should be typedef'ed as Parent_DX for X = class_name """
-        # a case where device logic parent cant be established
-        no_parent_structure = 'struct{/*No exact Parent of the class*/}'
+    def get_parent(self, class_name):
+        """Returns parent class name or None if device-logic parent is not applicable"""
         if self.class_has_device_logic(class_name):
             has_objects_origins = self.get_has_objects_origin_names(class_name, include_root=True)
             if len(has_objects_origins) == 1:
-                parent_structure = 'D'+has_objects_origins[0]
+                return has_objects_origins[0]
             else:
-                parent_structure = no_parent_structure
+                return None  # many parents, so no single parent.
         else:
-            parent_structure = no_parent_structure
-        if DEBUG:
-            print(Fore.YELLOW
-                  + "get_parent_struct({0}) gives {1}".format(
-                      class_name, parent_structure)
-                  + Style.RESET_ALL)
-        return parent_structure
+            return None  # child class w/o DeviceLogic, so no device lo
+
+    def get_parent_struct(self, class_name):
+        """ Returns whatever should be typedef'ed as Parent_DX for X = class_name """
+        # a case where device logic parent cant be established
+        parent = self.get_parent(class_name)
+        if parent is None:
+            return 'struct{/*No exact Parent of the class*/}'
```

Comment:
This `struct {/*blah*/}` seems like C++ specific markup creeping in here. I'd thought DesignInspector was rather a language agnostic source of information, and transformation of DesignInspector information into the specifics of the final format (I guess C++ code here) is the job of the jinja templates.

What do you think ?

Reply:
yes, I agree with you. Thanks for noticing. Since this 'creeping-in-of-C++' was already released (this is just a rework) I'd create an improvement ticket to address in one of upcoming releases. Thank you.

Reply:
OK - sounds good. No other comments: Looks good